tests/blast/test_blast_output.h

```diff
@@ -0,0 +1,814 @@
+                                    m.alignStats.numGaps,
+                                    m.eValue,
+                                    m.bitScore);
+                }
+            }
+            if ((format == 1) && (custom <= 1))
+                writeFooter(file, context, BlastTabular());
+            break;
+        case 2: // iteratre over records
+            writeHeader(file, context, BlastTabular()); // noop for TABULARs
+            for (auto const & r : records)
+                writeRecord(file, context, r, BlastTabular());
+            writeFooter(file, context, BlastTabular());
+            break;
+        case 3: // formatted file out
+        {
+            BlastTabularOut<BlastIOContext<TScore, TString, p, h>> out(context, file, BlastTabular());
+            writeHeader(out); // noop for TABULARs
+            for (auto const & r : records)
+                writeRecord(out, r);
+            writeFooter(out);
+        } break;
+    }
+}
+
+template <typename TFile, typename TScore, typename TString, typename TFormat, BlastProgram p, BlastTabularSpec h>
+inline void
+test_blast_write_record_match(TFile & file,
+                              int const format,
+                              int const customFields,
+                              BlastIOContext<TScore, TString, p, h> & context,
+                              TFormat const &)
+{
+    typedef Align<String<AminoAcid>, ArrayGaps> TAlign;
+    typedef BlastMatch<CharString, CharString, uint32_t, TAlign> TBlastMatch;
+    typedef BlastRecord<CharString, CharString, uint32_t, TAlign> TBlastRecord;
+
+    StringSet<String<AminoAcid>, Owner<ConcatDirect<>>> queries;
+    StringSet<CharString, Owner<ConcatDirect<>>> qIds;
+    StringSet<String<AminoAcid>> subjects;
+    StringSet<CharString> sIds;
+
+    resize(subjects, 2);
+    resize(sIds, 2);
+
+    subjects[0] =
+    "SSITEEKHIPHKEQDKDAEFLSKEALKTHMTENVLQMDRRAVQDPSTSFLQLLKAKGLLG"
+    "LPDYEVNLADVNSPGFRKVAYAQTKPRRLCFPNGGTRRGSFIMDTAVVVMVSLRYVNIGK"
+    "VIFPGATDVSEGEDEFWAGLPQAYGCLATEFLCIHIAIYSWIHVQSSRYDDMNASVIRAK"
+    "LNLAVITSWTQLIQAEKETI";
+
+    subjects[1] =
+    "GATRDSKGNAVITSFTQARLRVYADLLGPYWIILHVIELTGVGNTGQKCTLNHMGTYAVF"
+    "DLKQPPATNDLGLPKPCFIGFDIQNELAIGTVGHSEAVIAAFTQRDRLEERAESKQSLAR"
+    "PVISPKLIAEVSTVLESALNQMYSSLGFYRVERAEDYAQPRKLCVVKKKSFNCLNADIWL"
+    "EYRMEDQKSVPKVFKIMMDD";
+
+    sIds[0] = "Subject_Numero_Uno";
+    sIds[1] = "Subject_Numero_Dos";
+
+    appendValue(queries, "VAYAQPRKLCYP");
+    appendValue(queries, "NAYPRUTEIN");
+    appendValue(queries, "AVITSFTQ");
+
+    appendValue(qIds, "Query_Numero_Uno with args");
+    appendValue(qIds, "Query_Numero_Dos with args");
+    appendValue(qIds, "Query_Numero_Tres with args");
+
+    setScoreGapOpenBlast(context.scoringScheme, -11);
+    setScoreGapExtend(context.scoringScheme, -1);
+    SEQAN_ASSERT(isValid(context.scoringScheme));
+
+    String<TBlastRecord> records;
+    resize(records, 3);
+
+    context.dbName = "The Foo Database";
```

Comment:
Then we need to discuss these accessors because obviously they were not intended to be general. I don't have contigs, and I can't expect users to know that they have to say "contig" when they mean "protein in protein database".
[But I do agree that have the same accessors is very much desirable]